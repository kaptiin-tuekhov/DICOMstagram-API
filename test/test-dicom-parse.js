/* eslint import/no-extraneous-dependencies: "off",
no-use-extend-native/no-use-extend-native: "off" */
const fs = require('fs');
const Promise = require('bluebird');
const test = require('ava');
const dicomParse = require('../dicomParse');

Promise.promisifyAll(fs);

test('returns parsed dicom', async t => {
	const buff = await fs.readFileAsync('FluroWithDisplayShutter.dcm');
	t.snapshot(dicomParse(buff)); // eslint-disable-line ava/use-t-well
});
