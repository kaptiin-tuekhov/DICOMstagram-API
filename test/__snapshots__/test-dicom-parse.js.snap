exports[`returns parsed dicom 1`] = `"{\"SpecificCharacterSet\":\"ISO_IR 100\",\"ImageType\":\"ORIGINAL\\\\PRIMARY\\\\SINGLE PLANE\",\"InstanceCreationDate\":\"20070112\",\"InstanceCreationTime\":\"093126\",\"InstanceCreatorUID\":\"1.3.6.1.4.1.5962.3\",\"SOPClassUID\":\"1.2.840.10008.5.1.4.1.1.12.2\",\"SOPInstanceUID\":\"1.3.6.1.4.1.5962.1.1.0.0.0.1168612284.20369.0.3\",\"StudyDate\":\"20061207\",\"SeriesDate\":\"20061207\",\"AcquisitionDate\":\"20061207\",\"ContentDate\":\"20061207\",\"StudyTime\":\"091414\",\"SeriesTime\":\"091414.0157\",\"AcquisitionTime\":\"091414.0157\",\"ContentTime\":\"091414.0157\",\"AccessionNumber\":\"2\",\"Modality\":\"RF\",\"Manufacturer\":\"SIEMENS\",\"ReferringPhysicianName\":{\"dataOffset\":724,\"length\":0},\"TimezoneOffsetFromUTC\":\"-0500\",\"StudyDescription\":\"UPPER GI SERIES (STOMACH)\",\"ProcedureCodeSequence\":[{\"x00080100\":\"RADUGI\",\"x00080102\":\"BROKER\",\"x00080104\":\"UPPER GI SERIES (STOMACH)\",\"xfffee00d\":{\"dataOffset\":862,\"length\":0}}],\"x0008103e\":\"Barium        3F/s\",\"AdmittingDiagnosesDescription\":{\"dataOffset\":904,\"length\":0},\"ManufacturerModelName\":\"FLUOROSPOT_COMPACT\",\"PatientName\":\"Test^FluroWithDisplayShutter\",\"PatientID\":\"TEST2351267\",\"PatientBirthDate\":{\"dataOffset\":994,\"length\":0},\"PatientSex\":{\"dataOffset\":1002,\"length\":0},\"PatientIdentityRemoved\":\"YES\",\"DeidentificationMethod\":\"dcanon\\\\no identifiers\\\\keep descriptions\\\\keep manufacturer\\\\keep retired\\\\no private\\\\no uids\\\\no pet demographics\\\\keep dates\",\"ContrastBolusAgent\":{\"dataOffset\":1150,\"length\":0},\"KVP\":\"93\",\"SoftwareVersions\":\"VE20F\",\"DistanceSourceToDetector\":\"1150\",\"DistanceSourceToPatient\":\"972\",\"EstimatedRadiographicMagnificationFactor\":\"1.1831\",\"TableMotion\":\"STATIC\",\"TableAngle\":\"89\",\"FieldOfViewShape\":\"ROUND\",\"FieldOfViewDimensions\":\"300\",\"ExposureTime\":\"4\",\"XRayTubeCurrent\":\"325\",\"AveragePulseWidth\":\"4\",\"RadiationSetting\":\"GR\",\"x0018115a\":\"PULSED\",\"x0018115e\":\"0.3\",\"FilterType\":\"CU_0.0_MM\",\"IntensifierSize\":\"367\",\"ImagerPixelSpacing\":\"0.293\\\\0.293\",\"ColumnAngulation\":\"0\",\"ShutterShape\":\"RECTANGULAR\\\\CIRCULAR\",\"ShutterLeftVerticalEdge\":\"233\",\"ShutterRightVerticalEdge\":\"789\",\"ShutterUpperHorizontalEdge\":\"5\",\"ShutterLowerHorizontalEdge\":\"1018\",\"CenterOfCircularShutter\":\"512\\\\512\",\"RadiusOfCircularShutter\":\"517\",\"x0020000d\":\"1.3.6.1.4.1.5962.1.1.0.0.0.1168612284.20369.0.1\",\"x0020000e\":\"1.3.6.1.4.1.5962.1.1.0.0.0.1168612284.20369.0.2\",\"StudyID\":\"2\",\"SeriesNumber\":\"1\",\"AcquisitionNumber\":\"1\",\"InstanceNumber\":\"1\",\"PatientOrientation\":{\"dataOffset\":1652,\"length\":0},\"Laterality\":{\"dataOffset\":1660,\"length\":0},\"SamplesPerPixel\":\"1\",\"PhotometricInterpretation\":\"MONOCHROME2\",\"Rows\":\"1024\",\"Columns\":\"1024\",\"PixelSpacing\":\"0.293\\\\0.293\",\"BitsAllocated\":\"16\",\"BitsStored\":\"10\",\"HighBit\":\"9\",\"PixelRepresentation\":\"0\",\"PixelIntensityRelationship\":\"LIN\",\"WindowCenter\":\"640\",\"WindowWidth\":\"588\",\"RequestingPhysician\":{\"dataOffset\":1814,\"length\":0},\"CurrentPatientLocation\":\"RAD\",\"PerformedProcedureStepDescription\":\"UPPER GI SERIES (STOMACH)\",\"x7fe00010\":{\"dataOffset\":1872,\"length\":2097152},\"FileMetaInformationGroupLength\":\"194\",\"FileMetaInformationVersion\":{\"dataOffset\":156,\"length\":2},\"MediaStorageSOPClassUID\":\"1.2.840.10008.5.1.4.1.1.12.2\",\"MediaStorageSOPInstanceUID\":\"1.3.6.1.4.1.5962.1.1.0.0.0.1168612284.20369.0.3\",\"TransferSyntaxUID\":\"1.2.840.10008.1.2.1\",\"ImplementationClassUID\":\"1.3.6.1.4.1.5962.2\",\"ImplementationVersionName\":\"DCTOOL100\",\"SourceApplicationEntityTitle\":\"CLUNIE1\"}"`;
